src/common/libzmqutil/cert.c

```diff
@@ -0,0 +1,396 @@
+/************************************************************\
+ * Copyright 2023 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* cert.c - manage zeromq curve certificates
+ *
+ * Note that ZeroMQ curve certs generated by CZMQ use the ZPL config
+ * format.  Flux no longer uses CZMQ but retains compatibility with certs
+ * generated by earlier Flux versions that did.  Although Flux certs remain
+ * in ZPL form, this cert parser is far more restrictive, and therefore
+ * simpler, than the generic ZPL parser in CZMQ.
+ */
+
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <zmq.h>
+#include <string.h>
+
+#include "src/common/libutil/errno_safe.h"
+#include "src/common/libutil/strstrip.h"
+#include "src/common/libutil/strlcpy.h"
+#include "src/common/libczmqcontainers/czmq_containers.h"
+#include "ccan/str/str.h"
+
+#include "sockopt.h"
+#include "cert.h"
+
+#define KEYSIZE 32
+#define TXTSIZE 41
```

Comment:
A comment of where these sizes are coming from might be helpful. I'm guessing cert keys are limited to 32 bytes (256 bit) and the max expanded size of 32 bytes with z85 is 41 bytes?